### PR TITLE
ALTER level_sources id column from int to bigint

### PR DESCRIPTION
**Pull request by sureshc**
Use a Rails migration to alter the dashboard `level_sources` table `id` column from `int` to `bigint` before auto increment exceeds that maximum `int` (~2B). This migration does not run on production. Use `gh-ost` on production (#41927 ).

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/db/schema.rb [hunk 1]
```diff
@@ -301,7 +301,7 @@
     t.index ["follower_id"], name: "index_code_review_group_members_on_follower_id"
   end
 
-  create_table "code_review_groups", options: "ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci", force: :cascade do |t|
+  create_table "code_review_groups", options: "ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci", force: :cascade do |t|
```
Comment (Erin007) [inline dashboard/db/schema.rb]: Are you sure we want the other changes in to this file? I'd expect to see just the ones related to the migration on line 688.

**Comment by Sam-Hutchins** [comment]
Two questions:
1. Will this have any impact on the DMS task?
2. Does this automatically update the data type of foreign keys in the production database/does it need to?

File: dashboard/db/schema.rb (hunk 1 above)
Comment (sureshc) [reply dashboard/db/schema.rb]: Thanks for catching that! Some discrepancies between my local database configuration and our managed environments’ databases caused this. I’ve reverted those errant schema changes.

**Comment by sureshc** [comment]
> 1. Will this have any impact on the DMS task?

The production database `level_sources.id` column was already modified from `int` to `bigint` via a utility and is being exported to Redshift. Apologies for not giving you a heads up about that. I’m fairly certain, based on similar past datatype modifications that this won’t impact RED team.

> 2. Does this automatically update the data type of foreign keys in the production database/does it need to?

About 3 years ago, Ben implemented several changes to associated tables to use the new datatype. This parent table was the last to convert!

**Pull request merged**